update.sh

```diff
@@ -111,65 +204,108 @@ rm -f etc/maglev.demo.key
 ln -sf maglev.demo.key-$PLATFORM etc/maglev.demo.key
 # Make sure we have specs and benchmarks.
 echo "[Info] updating MSpec and RubySpec submodules"
-git submodule --quiet update --init
+ensure git submodule --quiet update --init
 
-# Create a default repository called "maglev" and generate the MagLev HTML documentation
 # Check for existence of required executable rake
-if [  -e "`which rake 2>/dev/null`" ]; then
-    if [ "$(dirname `which rake`)" == "${MAGLEV_HOME}/bin" ]; then
-        echo "[Error] Cannot use the rake in \$MAGLEV_HOME/bin to update. Please put another Ruby's rake in your PATH."
-        exit 1
-    fi
+if
+  which rake >/dev/null 2>&1
```

Comment:
Style-wise, do we want a space there? `which rake > /dev/null`? I've never seen a script w/o a space.